sim/src/main/scala/spinal/sim/VerilatorBackend.scala

```diff
@@ -460,15 +483,59 @@ JNIEXPORT void API JNICALL ${jniPrefix}disableWave_1${uniqueId}
        | --Mdir ${workspaceName}
        | --top-module ${config.toplevelName}
        | $rtlIncludeDirsArgs
-       | -cc ${config.rtlSourcesPaths.filter(e => e.endsWith(".v") || 
-                                                  e.endsWith(".sv") || 
-                                                  e.endsWith(".h"))
-                                     .map(new File(_).getAbsolutePath)
-                                     .map('"' + _.replace("\\","/") + '"')
-                                     .mkString(" ")}
+       | -cc ${verilatorInputFiles.map('"' + _.replace("\\","/") + '"').mkString(" ")}
        | --exe $workspaceName/$wrapperCppName
        | ${config.simulatorFlags.mkString(" ")}""".stripMargin.replace("\n", "")
 
+
+    if (cacheEnabled) {
+      // calculate hash of verilator version+options and source file contents
+
+      val md = MessageDigest.getInstance("SHA-1")
+      md.update(verilatorScript.getBytes())
+
+      val verilatorVersionProcess = Process(Seq(verilatorBinFilename, "--version"), new File(workspacePath))
+      val verilatorVersion = verilatorVersionProcess.lineStream.mkString("\n")    // blocks and throws an exception if exit status != 0
+      md.update(verilatorVersion.getBytes())
+
+      verilatorInputFiles.foreach { filename =>
```

Comment:
`verilatorInputFiles` is defined as the input source files for verilator. So this includes all `*.v`, `*.sv` and `*.h` files but not `*.mem` or `*.bin` files. Im not sure if that's really correct?

Reply:
verilatorInputFiles => file that are translated by verilator
config.rtlSourcesPaths.filter(s => s.endsWith(".bin") || s.endsWith(".mem"))    =>  files which are used as ressources.
config.rtlSourcesPaths => everytiong

Maybe you should use  config.rtlSourcesPaths instead of verilatorInputFiles ?

Reply:
done